omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -783,32 +896,33 @@ def _rotate_in_place(self, end_pose, angle_threshold = DEFAULT_ANGLE_THRESHOLD):
             
         yield self._empty_action()
             
-    def _sample_pose_near_object(self, obj, pos_on_obj=None, **kwargs):
-        if pos_on_obj is None:
+    def _sample_pose_near_object(self, obj, pose_on_obj=None, **kwargs):
+        if pose_on_obj is None:
             pos_on_obj = self._sample_position_on_aabb_face(obj)
+            pose_on_obj = np.array([pos_on_obj, [0, 0, 0, 1]])
+
+        with UndoableContext(self.robot, "base") as context:
```

Comment:
Why are we doing an undoablecontext here?

Reply:
faster way to test poses of when sampling